src/pkcs1v15.rs

```diff
@@ -774,6 +824,46 @@ impl EncryptingKeypair for DecryptingKey {
     }
 }
 
+mod oid {
+    use const_oid::ObjectIdentifier;
+
+    /// A trait which associates an RSA-specific OID with a type.
+    pub(crate) trait RsaSignatureAssociatedOid {
+        /// The OID associated with this type.
+        const OID: ObjectIdentifier;
+    }
+
+    #[cfg(feature = "sha1")]
+    impl RsaSignatureAssociatedOid for sha1::Sha1 {
+        const OID: ObjectIdentifier =
+            const_oid::ObjectIdentifier::new_unwrap("1.2.840.113549.1.1.5");
+    }
+
+    #[cfg(feature = "sha2")]
+    impl RsaSignatureAssociatedOid for sha2::Sha224 {
+        const OID: ObjectIdentifier =
+            const_oid::ObjectIdentifier::new_unwrap("1.2.840.113549.1.1.14");
```

Comment:
```suggestion
            const_oid::db::rfc5912::SHA_224_WITH_RSA_ENCRYPTION
```

Reply:
Yes, this was bugging me too. But I think the overall design now is not to add a dependency on `const-oid/db`.
@tarcieri ?